superset/views/database/forms.py

```diff
@@ -39,6 +40,12 @@
 )
 from superset.models.core import Database
 
+
+class DatabasesNotAllowedToFileUpload(str, Enum):
+    GSHEETS = "gsheets"
+    CLICKHOUSE = "clickhousedb"
```

Comment:
Rather than maintaining a separate enum for this, why not define this as a property on the db_engine_spec?

Reply:
Hey @nytai! I just updated the PR with such property in our `base` and each engine now. Thanks for the suggestion.

Reply:
@nytai let us know if you want us to recruit additional folks for a review, or if you have time to take a look. Thanks!

Reply:
I think having someone with more experience with these per-db configs, like @villebro or @john-bodley, have a look would be good.

Reply:
@Antonio-RiveroMartnez Thanks for making that change, multiple places in the frontend where you're checking 
`db?.engine !== Engines.GSheet && db?.engine !== Engines.ClickHouse`. This makes adding a new engine that does not support file upload a pain, since you have to update a bunch of files across both backend and frontend code. Ideally one could just specify in the db_engine_spec  `allows_file_upload=False` and that's it. So maybe adding a new api that you could call from the db modal to check is a supplied connection string or config supports file uploads would be a better approach?

Reply:
I agree with @nytai here - we should place this flag in the db engine specs, and the UI should accommodate accordingly. I may be wrong, but I believe we're already sending some db engine spec metadata in bootstrap data. Maybe we could just add `allows_file_upload` to `BaseEngineSpec`, override it in the GSheets and ClickHouse specs, and then make sure the flag is made available in bootstrap data? If we're not passing any info about db engine specs, now would probably be a good time to do so.

Reply:
Ok @nytai @villebro , I updated the PR, we were sending some `extra` information in the response but that might is editable by the user so I decided to include a new `engine_information` that currently only has `allows_file_upload` flag. Now, There is no need for updating multiple places but instead just setting such flags as true to prevent file uploading.